sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -311,11 +313,12 @@ private[hive] class HiveClientImpl(
   override def createDatabase(
       database: CatalogDatabase,
       ignoreIfExists: Boolean): Unit = withHiveState {
+    // default database's location always use the warehouse path, here set to emtpy string
     client.createDatabase(
       new HiveDatabase(
         database.name,
         database.description,
-        database.locationUri,
+        if (database.name == SessionCatalog.DEFAULT_DATABASE) "" else database.locationUri,
```

Comment:
sorry, actually it will throw an exception, my local default has created, so it does not hit the exception, I will just replace the default database location when reload from metastore, drop the logic when create database set location to empty string.